#!/usr/bin/env python3
PACKAGE = "cube_tracker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("core_point_min_",        int_t,       0,   "core_point_min_",      7,      0,     10000)
gen.add("cluster_tolerance_",     double_t,    0,   "cluster_tolerance_",   0.4,     0,     1.0)
gen.add("min_cluster_size_",      int_t,       0,   "min_cluster_size_",    10,      0,     10000)
gen.add("max_cluster_size_",      int_t,       0,   "max_cluster_size_",    1000,     0,     10000)
gen.add("voxel_size_",            double_t,    0,   "voxel_size",          0.15,     0,    10)
gen.add("filter_min_x",            int_t,       0,   "filter_min_x",          -70,     -100,   100)
gen.add("filter_max_x",            int_t,       0,   "filter_max_x",          70,      -100,   100)
gen.add("filter_min_y",            int_t,       0,   "filter_min_y",          -70,     -100,   100)
gen.add("filter_max_y",            int_t,       0,   "filter_max_y",          70,      -100,   100)
gen.add("filter_min_z",            int_t,       0,   "filter_min_z",          -2,     -100,   100)
gen.add("filter_max_z",            int_t,       0,   "filter_max_z",          10,      -100,   100)
gen.add("volume", double_t,     0,    "volume",   20,    0,    100)

exit(gen.generate(PACKAGE, "cube_tracker", "SetClusterParam"))